### PR TITLE
Alter bazel-test-canary to run on k8s-infa-prow-build

### PR DESCRIPTION
**Pull request by spiffxp**
Need to make some changes to do so:
- stop using RBE
- take a guess at resource limits (will likely need tuning since the
  guess is based on runs that used RBE)

### CODE REVIEW

**Comment by spiffxp** [comment]
/cc @hasheddan @liggitt @BenTheElder 
prep for https://github.com/kubernetes/test-infra/issues/18582

File: config/jobs/kubernetes/sig-testing/bazel-build-test.yaml [hunk 1]
```diff
@@ -303,11 +303,13 @@ periodics:
...
       - -//build/...
       - -//vendor/...
+
 - name: periodic-kubernetes-bazel-test-canary
+  cluster: k8s-infra-prow-build
   annotations:
     testgrid-dashboards: sig-testing-canaries
     testgrid-tab-name: bazel-test
-    description: run kubernetes-bazel-test with latest image
+    description: run kubernetes-bazel-test without RBE on k8s-infra-prow-build
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-testing/bazel-build-test.yaml:312]: is "without RBE" necessary? only test-infra is using it currently, nominally

Comment (spiffxp) [reply config/jobs/kubernetes/sig-testing/bazel-build-test.yaml:312]: yes, there are kubernetes/kubernetes jobs that use RBE either via `--config=ci` or `--config=remote` (ref: https://github.com/kubernetes/kubernetes/blob/master/build/root/.bazelrc#L52-L87)

e.g.
- periodic-bazel-test-master above
- periodic-bazel-build-based

**Comment by spiffxp** [comment]
Jobs that failed due to pod scheduling timeout since this merged:
- https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/periodic-kubernetes-bazel-test-canary/1290126399931158528
- https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/periodic-kubernetes-bazel-test-canary/1290203156529025025
- https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/periodic-kubernetes-bazel-test-canary/1290279911495307264
- https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/periodic-kubernetes-bazel-test-canary/1290371879034425346
- https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/periodic-kubernetes-bazel-test-canary/1290387226114396164

None of them state what resource was disqualifying them from scheduling, next step is search the logs for k8s-infra-prow-build and look for references to the pods these jobs tried to schedule

**Comment by hasheddan** [comment]
@spiffxp I'm going to see if I can get that info presented in spyglass. However, I think it will require significant changes due to the fact that it isn't surfaced on the `prowjob.json` from what I can tell

**Comment by spiffxp** [comment]
Yeah I'm not clear on why the info isn't surfacing

**Comment by spiffxp** [comment]
Picking the most recent:
- https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/periodic-kubernetes-bazel-test-canary/1290387226114396164
- "Prow Job YAML"
- https://prow.k8s.io/prowjob?prowjob=a035e1aa-d5c9-11ea-b94d-968fdbb2a550
- `pod_name: a035e1aa-d5c9-11ea-b94d-968fdbb2a550`
- [Logging for resource Kubernetes Pod with text a035e1aa-d5c9-11ea-b94d-968fdbb2a550](https://console.cloud.google.com/logs/viewer?project=k8s-infra-prow-build&minLogLevel=0&expandAll=false&timestamp=2020-08-03T23:18:51.975000000Z&customFacets=&limitCustomFacetWidth=true&dateRangeStart=2020-08-03T17:18:53.085Z&dateRangeEnd=2020-08-03T23:18:53.085Z&interval=PT6H&resource=k8s_pod&scrollTimestamp=2020-08-03T20:42:03.000000000Z&filters=text:a035e1aa-d5c9-11ea-b94d-968fdbb2a550&angularJsUrl=%2Flogs%2Fviewer%3Fproject%3Dk8s-infra-prow-build%26minLogLevel%3D0%26expandAll%3Dfalse%26timestamp%3D2020-08-03T23:17:11.669000000Z%26customFacets%3D%26limitCustomFacetWidth%3Dtrue%26dateRangeStart%3D2020-08-03T17:17:11.923Z%26dateRangeEnd%3D2020-08-03T23:17:11.923Z%26interval%3DPT6H%26resource%3Dk8s_pod%26scrollTimestamp%3D2020-08-03T20:42:03.000000000Z%26filters%3Dtext:a035e1aa-d5c9-11ea-b94d-968fdbb2a550)
- 2020-08-03 13:41:03.000 PDT "0/22 nodes are available: 12 Insufficient memory, 19 Insufficient cpu, 3 node(s) had taints that the pod didn't tolerate."
- 2020-08-03 13:41:08.000 PDT "pod triggered scale-up:"
- 2020-08-03 13:42:03.000 PDT "skip schedule deleting pod: test-pods/a035e1aa-d5c9-11ea-b94d-968fdbb2a550"

questions this raises:
- when did the scale-up finish?
- was the pod deleted before the scale-up finished?
- why was the pod deleted?

**Comment by spiffxp** [comment]
https://console.cloud.google.com/monitoring/dashboards/custom/10925237040785467832?project=k8s-infra-prow-build&startTime=20200720T162400-07:00&endTime=20200803T162416-07:00 - well, we're definitely scaling up/down, which is a sign that the resource requests/limits are helping

**Comment by spiffxp** [comment]
Logs from k8s-prow (google.com) project
```
2020-08-03 13:40:54.000 PDT - horologium - Triggering new run of interval periodic.
2020-08-03 13:41:03.000 PDT - plank - Create Pod.
...
2020-08-03 13:41:15.000 PDT - crier - Successfully updated prowjob (jobStatus: pending)
2020-08-03 13:42:03.000 PDT - plank - Marked job for stale unscheduled pod as errored
2020-08-03 13:42:03.000 PDT - plank - Delete stale running pod
2020-08-03 13:42:04.000 PDT - crier - Successfully updated prowjob (jobStatus: error)
...
2020-08-03 13:55:07.000 PDT - sinker - Deleted prowjob.
```

So prow's the one doing the deleting.  Who do I have to tell to wait longer?

**Comment by spiffxp** [comment]
"marked job for stale" comes from https://github.com/kubernetes/test-infra/blob/86ee5af7acbc989c90fca4edb1cf61afd7cec8b4/prow/plank/controller.go#L390-L405

**Comment by spiffxp** [comment]
And maxPodUnscheduled comes from...
https://github.com/kubernetes/test-infra/blob/86ee5af7acbc989c90fca4edb1cf61afd7cec8b4/config/prow/config.yaml#L7-L8

OK, so I think 1m is too aggressive if we need to see scale-up succeed, the question is how long is reasonable?

**Comment by spiffxp** [comment]
How did we decide on 1m? https://github.com/kubernetes/test-infra/pull/16089 says "I expect the pod_unscheduled_timeout change to have no effect: pods are always scheduled in under one second, or never scheduled at all."

Which was before we scheduled to clusters that autoscale

**Comment by spiffxp** [comment]
k8s-infra-prow-build - Kubernetes Cluster for text "gke-prow-build-pool1-2020043022092218-c3efd08d"
```
2020-08-03 13:41:06.000 PDT - Scale-up: setting group https://content.googleapis.com/compute/v1/projects/k8s-infra-prow-build/zones/us-central1-f/instanceGroups/gke-prow-build-pool1-2020043022092218-c3efd08d-grp size to 7
2020-08-03 13:41:08.000 PDT - Scale-up: group https://content.googleapis.com/compute/v1/projects/k8s-infra-prow-build/zones/us-central1-f/instanceGroups/gke-prow-build-pool1-2020043022092218-c3efd08d-grp size set to 7
2020-08-03 13:42:40.805 PDT - protoPayload.methodName: "io.k8s.core.v1.nodes.create", 
```

start looking for text "gke-prow-build-pool1-2020043022092218-c3efd08d-plvr"

logs settle to a bunch of "kube-node-lease" updates by 13:44

looked for a tighter bound by looking for `protoPayload.response.status.conditions.status="True"`, last entry by 13:42:49

tl;dr
- took 2-3min from scale-up to node ready
- I'd like to suggest 5min to give us room

**Comment by spiffxp** [comment]
Opened https://github.com/kubernetes/test-infra/pull/18637 to propose a longer timeout
